chapter09/red/exercise3.pl

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use Regexp::Assemble;
+
+my ( $pattern_repo ) = @ARGV;
+if ( !defined $pattern_repo || !-r $pattern_repo ) {
+  print "Please supply a file containing line delimited regular expressions.\n";
+  exit 0;
+}
+
+my $regex = eval { new Regexp::Assemble( file => $pattern_repo )->re };
+die "Invalid regular expression: $@" if $@;
+
```

Comment:
True enough: `Invalid regular expression: cannot open afjioajefoj for input: No such file or directory at -e line 1`. Still, the user gets some kind of error, just not a wholly accurate one :wink:.